app.js

```diff
@@ -29,33 +30,59 @@ const sessionParms = {
 };
 
 if (app.get("env") === "production") {
-  app.set("trust proxy", 1); // trust first proxy
-  sessionParms.cookie.secure = true; // serve secure cookies
+  app.set("trust proxy", 1);
+  sessionParms.cookie.secure = true;
 }
 
+app.use(cookieParser()); // Must be before CSRF middleware
 app.use(session(sessionParms));
 
 const passport = require("passport");
 const passportInit = require("./passport/passportInit");
-
 passportInit();
 app.use(passport.initialize());
 app.use(passport.session());
 
 app.use(require("connect-flash")());
-
 app.use(require("./middleware/storeLocals"));
+
+//  CSRF Protection Middleware (Must be after session and body-parser)
+app.use(csurf());
+
+//  Middleware to pass CSRF token to views
+app.use((req, res, next) => {
+  res.locals._csrf = req.csrfToken();  // Make sure CSRF token is available in views
+  next();
+});
+
+// Jobs - Add New Job Page
+app.get("/jobs/new", (req, res) => {
+  res.render("job", { job: null, _csrf: req.csrfToken() }); // Explicitly pass job as null
```

Comment:
For example, this should function the same like so:
// ...
res.render("job", { job: null });
// ...